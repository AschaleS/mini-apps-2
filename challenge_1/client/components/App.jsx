import React from 'react';
import axios from 'axios';
import EventList from './EventList.jsx';
import ReactPaginate from 'react-paginate';
import './style.scss';


class App extends React.Component {
  constructor() {
    super();

    this.state = {
      term: '',
      results: [],
      totalCount: 0
    }

    this.getData = this.getData.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleChange(e) {
    this.setState({
      term: e.target.value
    })
  }

  getData() {
    axios.get(`/events?q=${this.state.term}&_page=1&_limit=10`)
      .then((response) => {
        let count = response.headers["x-total-count"];
        this.setState({
          results: response.data,
          totalCount: count
        })
      })
      .catch((error) => {
        console.log(error);
      });
  }



  handleClick(event) {
    axios.get(`/events?q=${this.state.term}&_page=${event.selected + 1}&_limit=10`)
      .then((response) => {
        this.setState({
          results: response.data
        })
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    if (this.state.results.length === 0) {
      return (
        <div style={{ color: "#511904", padding: "35px", fontWeight: "bold", background: "#0b96a8", textAlign: "center" }}>
          <h1>Historical Events Finder</h1>
          <div style={{ color: "#040e13", padding: "35px", fontWeight: "bold", fontSize: "30px", textAlign: "left" }}>

            Enter a serach key here:<input style={{ width: "300px", height: "30px", fontSize: "22px" }} onChange={this.handleChange} />
            <button style={{ fontSize: "14px", fontWeight: "bold", height: "35px" }} onClick={this.getData}> Get Events </button>

          </div>
        </div>)
    } else {
      return (
        <div>
          <div>
            <div style={{ color: "#511904", padding: "35px", fontWeight: "bold", background: "#0b96a8", textAlign: "center" }}>
              <h1>Historical Events Finder</h1>
              <div style={{ color: "#040e13", padding: "35px", fontWeight: "bold", fontSize: "30px", textAlign: "left" }}>

                Enter a serach key here:<input style={{ width: "300px", height: "30px", fontSize: "22px" }} onChange={this.handleChange} />
                <button style={{ fontSize: "14px", fontWeight: "bold", height: "35px" }} onClick={this.getData}> Get Events </button>

              </div>
            </div >
            <div style={{ padding: "15px" }}>
              < EventList results={this.state.results} />
            </div>

          </div>
          <div id="react-paginate">
            <ReactPaginate
              previousLabel={'previous'}
              nextLabel={'next'}
              breakLabel={'...'}
              breakClassName={'break-me'}
              pageCount={Math.ceil(this.state.totalCount / 10)}
              marginPagesDisplayed={2}
              pageRangeDisplayed={5}
              onPageChange={this.handleClick}
              containerClassName={'pagination'}
              activeClassName={'active'}
            />
          </div>


        </div>



      )
    }

  }
}

export default App;
