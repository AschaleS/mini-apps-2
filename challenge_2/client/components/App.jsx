import React from 'react';
import ReactDOM from 'react-dom'
import axios from 'axios';



class App extends React.Component {
  constructor() {
    super();

    this.state = {
      date: [],
      price: []
    }
  }
  componentDidMount() {
    axios.get('/priceData')
    .then((response) => {
      console.log(response)
      this.setState({
       date: Object.keys(response.data.bpi),
       price: Object.values(response.data.bpi)
      });
    })
    .catch((error) => {
      console.log(error);
    });
  }



  render() {

      return (
        <div>
          <h1 style={{textAlign: "center"}}>Cryptocurrency Charting Tool</h1>
                    <p style={{textAlign: "center"}}>Chart Data will be displayed here</p>
          <h4 style={{textAlign: "center"}}>Powered by CoinDesk</h4>
        </div>
      )
    }
}
export default App;
