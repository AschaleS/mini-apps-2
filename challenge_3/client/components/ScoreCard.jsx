import React from 'react';

const ScoreCard = (props) => {
  if (props.score.frame2 === 2) {
    return (
      <div style={{fontSize: "18px", fontWeight: "bold"}}>
        <br/><div> # of Pins Hit: {props.score.firstBowl}</div>
        <div> Frame: {props.score.frame}</div>
        <div> Total Score: {props.score.score}</div>
      </div>
    )
  } else {
    return (
      <div style={{fontSize: "18px", fontWeight: "bold"}}>
        <br/><div> # of Pins Hit: {props.score.secondBowl}</div>
        <div> Frame: {props.score.frame}</div>
        <div> Total Score: {props.score.score}</div>
      </div>
    )
  }
}

export default ScoreCard;
