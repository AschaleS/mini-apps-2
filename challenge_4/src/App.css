.App {
  text-align: center;
}


.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: rgb(10, 157, 64);
  height: 20px;
  padding: 15px;
  color: white;
}

.App-title {
  font-size: 1.2em;
}

.App-intro {
  font-size: large;
  color: blue;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}


.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}


form {
  margin: 30px 0;
}

select {
  padding: 5px 10px;
  margin-right: 10px;
  font-size: 1.0em;
  border-width: 1px;
  border-color: #e1e4e2;
  background-color: #6cb3da;
}

table {
  margin: 0 auto;
  table-layout: fixed;
  margin-top: 75px;
  border-style: inset;
  border-width: 1px;
  border-color: #cbcbcb #fff #fff #cbcbcb;
  background-color: #ded9d8;
}

table tbody {
  border-collapse: collapse;
  display: table-cell;
}


table td {
  font-weight: normal;
  height: 38px;
  width: 38px;
  padding: 0;
  box-sizing: content-box;
  border: 1px solid rgb(27, 69, 96);
  font-family: 'Arial Black';
  font-size: 1.2em;
}

button {
  border-style: outset;
  border-width: 3px;
  border-color: #e1e4e2 #cbcbcb #cbcbcb #fff;
  background-color: #dad4f7;
  outline: 0;
  cursor: pointer;
  padding:0;
}

form button {
  padding: 5px 10px;
  font-size: 1.2em;
}

td button {
  height: 103%;
  width: 103%;
}

.bomb {
  background-color: rgb(247, 79, 79);
}

.game-over,
.win {
  position: absolute;
  top: 115px;
  left: 50%;
  width: 300px;
  margin-left: -150px;
  box-sizing: border-box;
  font-family: 'Ubuntu';
  color: rgb(0, 0, 255);
  font-size: 1.8em;
}

.one {
  color: #d75252;
}

.two {
  color: #0a7916;
}

.three {
  color: #0000ff8a;
}

.four {
 color: #03a9bbd9;
}

