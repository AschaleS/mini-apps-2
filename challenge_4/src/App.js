import React from 'react';
import BoardContainer from '../src/containers/BoardContainers.js'
import './App.css';


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      boardSize: '',
    };
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render() {
    const { onPlayAgain, gameover, counter, win } = this.props;
    const { boardSize} = this.state;
    return (
      <div className="App">
        <h2 style={{color: "#15a52c", fontFamily: "American Typewriter"}}>Play by finding all the boxes without mines! </h2>

         <form onSubmit={(e) => {
          e.preventDefault();
          onPlayAgain(boardSize);
        }}>
           <button type="submit">New Game</button>
        </form>
        {gameover &&
          <div className="game-over">Game over! You Just Lost!</div>
        }
        {win &&
          <div className="win">Congrats, you won the Game!</div>
        }
        {counter > -1 &&
          <div>
            <BoardContainer />
          </div>
        }
      </div>
    );
  }
}

export default App;

