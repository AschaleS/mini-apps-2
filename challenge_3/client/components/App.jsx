import React from 'react';
import ReactDOM from 'react-dom'

import ScoreCard from './ScoreCard.jsx';



class App extends React.Component {
  constructor() {
    super();

    this.state = {
      frame: 1,
      frame2: 1,
      firstBowl: 0,
      secondBowl: 0,
      score: 0,
      strikeBonus: 0,
      spareBonus: 0
    }
    this.pinSelection = this.pinSelection.bind(this);
  }

  pinSelection(e) {
    var points = Number(e.target.value);

    if (this.state.strikeBonus > 3) {
      this.setState({
        strikeBonus: 0
      })
    }
    if (this.state.spareBonus > 2) {
      this.setState({
        spareBonus: 0
      })
    }
    this.setState({
      frame2: this.state.frame2 + 1,
      score: this.state.score + points,
    })
    if (this.state.frame2 === 1) {
      if (this.state.strikeBonus === 1) {
        this.setState({
          score: this.state.score + (points * 2),
          strikeBonus: this.state.strikeBonus + 1
        })
      }
      if (this.state.spareBonus === 1) {
        this.setState({
          score: this.state.score + (points * 2),
          spareBonus: this.state.spareBonus + 1
        })
      }
      this.setState({
        firstBowl: points
      })
    }
    if (this.state.frame2 === 2) {
      if (this.state.strikeBonus === 2) {
        this.setState({
          score: this.state.score + (points * 2),
          strikeBonus: this.state.strikeBonus + 1
        })
      }
      this.setState({
        secondBowl: points,
        frame: this.state.frame + 1,
        frame2: 1
      })
    }

    if (points === 10 && this.state.frame2 === 1) {
      this.setState({
        frame: this.state.frame + 1,
        frame2: 1,
        strikeBonus: this.state.strikeBonus + 1
      })
    }

    if (this.state.firstBowl + points === 10) {
      this.setState({
        spareBonus: 1
      })
    }

    if (this.state.frame === 10) {
      alert('The Game is Over! Your Total Score is: ' + this.state.score);
      this.setState({
        frame: 1,
        frame2: 1,
        firstBowl: 0,
        secondBowl: 0,
        score: 0,
        strikeBonus: 0,
        spareBonus: 0
      })
    }
  }



  render() {

    return (

      <div style={{alignItems: "center", padding: "25px"}}>
        <h2>Bowling</h2>
      <div>
        <button value='1' onClick={(e) => this.pinSelection(e)}>1</button>
        <button value='2' onClick={(e) => this.pinSelection(e)}>2</button>
        <button value='3' onClick={(e) => this.pinSelection(e)}>3</button><br/>
        <button value='4' onClick={(e) => this.pinSelection(e)}>4</button>
        <button value='5' onClick={(e) => this.pinSelection(e)}>5</button>
        <button value='6' onClick={(e) => this.pinSelection(e)}>6</button><br/>
        <button value='7' onClick={(e) => this.pinSelection(e)}>7</button>
        <button value='8' onClick={(e) => this.pinSelection(e)}>8</button>
        <button value='9' onClick={(e) => this.pinSelection(e)}>9</button><br/>
        <button value='10' onClick={(e) => this.pinSelection(e)}>10</button>
        <button value='0' onClick={(e) => this.pinSelection(e)}>0</button>
        <button value='*'>*</button>
        <ScoreCard score={this.state} />
        </div>

      </div>
    )
  }
}
export default App;
