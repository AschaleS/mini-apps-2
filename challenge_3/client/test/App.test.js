import React from 'React';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from '../components/App.jsx';
import ScoreCard from '../components/ScoreCard.jsx';


configure({adapter: new Adapter()});

describe('App', () => {
  it('should renders App correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });


    it("should includes one div", () => {
      const wrapper = shallow(<App />);
      expect(wrapper.find('div').length).toEqual(1);
    });


    it("should render ScoreCard component", () => {
      const wrapper = shallow(<App />);
      const scorecard = wrapper.find(ScoreCard);
      expect(scorecard.exists()).toBe(true);
    });
});

