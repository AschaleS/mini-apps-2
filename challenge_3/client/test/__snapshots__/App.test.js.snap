// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `ShallowWrapper {}`;

exports[`App should renders App correctly 1`] = `ShallowWrapper {}`;
